File: ts/src/ordertaking/domain/order.ts
import {Undefined} from "../common/common";
import {Entity} from "../../libs/model-type";
import { OrderId } from "../type/types";

type CustomerInfo = Undefined;
type ShippingAddress = Undefined;
type BillingAddress = Undefined;
type OrderLine = Undefined;
type BillingAmount = Undefined;

class Order extends Entity<OrderId> {
    constructor(
        readonly customerInfo: CustomerInfo,
        readonly shippingAddress: ShippingAddress,
        readonly billingAddress: BillingAddress,
        readonly orderLines: OrderLine[],
        readonly amountToBill: BillingAmount
    ) {
        super()
    }
}

File: src/main/kotlin/ordertaking/domain/Order.kt
package com.rojojun.ordertaking.domain

import com.rojojun.ordertaking.common.Entity
import com.rojojun.ordertaking.common.Undefined

typealias CustomerInfo = Undefined
typealias ShippingAddress = Undefined
typealias BillingAddress = Undefined
typealias OrderLine = Undefined

class Order (
    val orderId: OrderId,
    val customerId: CustomerId,
    val orderLines: List<OrderLine>,
    val shippingAddress: ShippingAddress,
    val billingAddress: BillingAddress,
    val amountToBill: BillingAmount,
) : Entity<OrderId>() {
    override val id: OrderId
        get() = orderId
}
